api/apiclient_test.go

```diff
@@ -773,15 +906,68 @@ type fakeConn struct {
 	closed chan struct{}
 }
 
-func (c *fakeConn) Receive(x interface{}) error {
+func (c fakeConn) Receive(x interface{}) error {
 	return errors.New("no data available from fake connection")
 }
 
-func (c *fakeConn) Send(x interface{}) error {
+func (c fakeConn) Send(x interface{}) error {
 	return errors.New("cannot write to fake connection")
 }
 
-func (c *fakeConn) Close() error {
-	close(c.closed)
+func (c fakeConn) Close() error {
+	if c.closed != nil {
+		close(c.closed)
+	}
 	return nil
 }
+
+// fakeResolver implements IPAddrResolver
+// by looking up the addresses in the map,
+// which maps host names to IP addresses.
+type fakeResolver map[string][]string
+
+func (r fakeResolver) LookupIPAddr(ctx context.Context, host string) ([]net.IPAddr, error) {
+	if ip := net.ParseIP(host); ip != nil {
+		return []net.IPAddr{{IP: ip}}, nil
+	}
+	ipStrs := r[host]
+	if len(ipStrs) == 0 {
+		return nil, errors.Errorf("mock resolver cannot resolve %q", host)
+	}
+	ipAddrs := make([]net.IPAddr, len(ipStrs))
+	for i, ipStr := range ipStrs {
+		ip := net.ParseIP(ipStr)
+		if ip == nil {
+			panic("invalid IP address: " + ipStr)
+		}
+		ipAddrs[i] = net.IPAddr{
+			IP: ip,
+		}
+	}
+	return ipAddrs, nil
+}
+
+// seqResolver returns an implementation of
+// IPAddrResolver that maps the given addresses
+// to sequential IP addresses 0.1.1.1, 0.2.2.2, etc.
+func seqResolver(addrs ...string) api.IPAddrResolver {
+	r := make(fakeResolver)
+	for i, addr := range addrs {
+		host, _, err := net.SplitHostPort(addr)
+		if err != nil {
+			panic(err)
+		}
+		r[host] = []string{fmt.Sprintf("0.%[1]d.%[1]d.%[1]d", i+1)}
```

Comment:
odd side-note, github's golang renderer doesn't understand this syntax and flags the '%' characters as being an error.